src/bam/record.rs

```diff
@@ -113,12 +113,23 @@ fn extranul_from_qname(qname: &[u8]) -> usize {
 impl Record {
     /// Create an empty BAM record.
     pub fn new() -> Self {
```

Comment:
The goal in this PR is to that one does not need to call any other methods on `Record` to have a valid record.  Currently, this requires setting the read name and alignment positions (and unmapped flag if the positions are -1).  So this PR has an opinion, that a new record is unmapped with empty name/seq/qual/cigar.

Reply:
Thanks for indirectly making me think about `Default` trait vs `new()`: https://users.rust-lang.org/t/when-to-use-default-trait/11190

Reply:
Coincidental good read on Default: https://cj.rs/blog/rust-default-values-for-maintainability/